pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -2232,12 +2250,28 @@ public boolean next() throws SQLException {
       }
 
       // Execute the fetch and update this resultset.
-      connection.getQueryExecutor()
-          .fetch(cursor, new CursorResultHandler(), fetchRows, adaptiveFetch);
+      if (refCursorName == null || refCursorName.isEmpty()) {
+        castNonNull(resultCursor, "resultCursor");
+        connection.getQueryExecutor()
+            .fetch(resultCursor, new CursorResultHandler(), fetchRows, adaptiveFetch);
+        // .fetch(...) could update this.cursor, and cursor==null means
+        // there are no more rows to fetch
 
-      // .fetch(...) could update this.cursor, and cursor==null means
-      // there are no more rows to fetch
-      closeRefCursor();
+      } else {
+        StringBuilder sb = new StringBuilder("FETCH FORWARD ");
+        sb.append(fetchRows);
+        sb.append(" IN ");
```

Comment:
I am afraid this change does not fix the issue.

It uses `fetchRows` from the base `ResultSet` rather than `fetchRows` of the `ResultSet` that is associated with `refcursor` in question.

In other words:

Imagine we call `{? = call test_blob(?)}`.
`CallableStatement.execute` would gather call results, and it would execute `getObject` at https://github.com/pgjdbc/pgjdbc/blob/ca25d1053199da96a63838f7476a14137dc9d9be/pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java#L131

Then it would delegate to `PgResultSet.getObject -> internalGetObject -> getRefCursor`, and it would effectively use `fetchSize` from a temporary resultset associated with `{? = call test_blob(?)}`.

I believe we should add a test to ensure `fetch forward` is triggered at all.